csharp/src/Apache.Arrow/Ipc/ArrowReaderImplementation.cs

```diff
@@ -113,33 +115,35 @@ internal static ByteBuffer CreateByteBuffer(ReadOnlyMemory<byte> buffer)
             ByteBuffer messageBuffer,
             Flatbuf.RecordBatch recordBatchMessage)
         {
-            var arrays = new List<IArrowArray>(recordBatchMessage.NodesLength);
-            int bufferIndex = 0;
+            return CreateInner().ToList();
```

Comment:
Why change this to Enumerable.ToList()? The original code only allocated the list once up-front given the NodesLength capacity. Changing it to use Enumerable.ToList() means the underlying buffer needs to be reallocated as the list gets bigger.

Reply:
Thank you.
I had misunderstood the Enumerable.ToList() behaviour.
I'll modify it.